posts/inside-rust/2022-08-11-diagnostic-effort.md

```diff
@@ -0,0 +1,336 @@
+---
+layout: post
+title: "Contribute to the diagnostic translation effort!"
+author: David Wood
+team: the compiler team <https://www.rust-lang.org/governance/teams/compiler>
+---
+
+The Rust Diagnostics working group is leading an effort to add support
+internationalization of error messages in the compiler, allowing the compiler
+to produce output in languages other than English.
+
+For example, consider the following diagnostic where a user has used a colon to
+specify a function's return type instead of an arrow:
+
+```text
+error: return types are denoted using `->`
+ --> src/main.rs:1:21
+  |
+1 | fn meaning_of_life(): u32 { 42 }
+  |                     ^ help: use `->` instead
+```
+
+We could output that diagnostic in Chinese:
+
+```text
+错误: 返回类型使用`->`表示
```

Comment:
I'm mostly just curious but are these translations from a native speaker or from Google Translate or something? If the latter, it might be a good idea to double check them and make sure they don't sound too weird.

Reply:
I had colleagues who are native speakers provide these translations, so they should be good.